### PR TITLE
DM-32049: prevent the rotator following loop from trying to command the camera cable wrap with out-of-range position or velocity

### PR DESCRIPTION
**Pull request by r-owen**

### CODE REVIEW

File: python/lsst/ts/MTMount/limits.py [hunk 1]
```diff
@@ -60,6 +59,8 @@ def scaled(self, factor=1.01):
         )
 
 
+# Comand limits (the CSC does not actually command velocity or acceleration,
```
Comment (wvreeven) [inline python/lsst/ts/MTMount/limits.py:62]: `Comand` -> `Command`

File: tests/test_csc.py [hunk 2]
```diff
@@ -23,17 +23,22 @@
...
 import pathlib
 import logging
-import time
 import unittest
 
+import pytest
+
 from lsst.ts import salobj
 from lsst.ts import mtmount
+from lsst.ts import utils
```
Comment (wvreeven) [inline tests/test_csc.py:32]: A bit of nitpicking but wouldn't `from lsst.ts import salobj, mtmount, utils` make sense here?

File: tests/test_csc.py [hunk 3]
```diff
@@ -329,213 +329,136 @@ async def test_camera_cable_wrap_following(self):
...
+                    position_limit = ccw_limits.max_position
+                    velocity_limit = ccw_limits.max_velocity
+                else:
+                    position_limit = ccw_limits.min_position
+                    velocity_limit = -ccw_limits.max_velocity
+                velocity = velocity_limit * 1.1
+                position0 = position_limit - (velocity * trunc_time)
+                print(
+                    f"Position limit={position_limit}, velocity limit={velocity_limit}"
                 )
```
Comment (wvreeven) [inline tests/test_csc.py:413]: Is this print useful for the unit test or a remainder of debugging the code?

File: tests/test_csc.py [hunk 4]
```diff
@@ -329,213 +329,136 @@ async def test_camera_cable_wrap_following(self):
...
-            self.assertFalse(mock_azimuth.tracking_enabled)
-            self.assertFalse(mock_elevation.tracking_enabled)
+                tai0 = utils.current_tai()
+                previous_tai = 0
+                while True:
+                    tai = utils.current_tai()
+                    # Work around non-monotonic clocks, which are
+                    # sometimes seen when running Docker on macOS.
+                    if tai < previous_tai:
+                        tai = previous_tai + 0.001
```
Comment (wvreeven) [inline tests/test_csc.py:422]: Is this really enough to ensure that `tai` is larger than `previous_tai`? Wouldn't establishing the difference and adding that plus a small margin to tai be a better approach?

File: tests/test_csc.py [hunk 5]
```diff
@@ -329,213 +329,136 @@ async def test_camera_cable_wrap_following(self):
...
+                    )
+                    if do_max_limit:
+                        assert ccw_target.velocity <= ccw_limits.max_velocity
+                        assert ccw_target.position <= position_limit
+                    else:
+                        assert ccw_target.velocity >= -ccw_limits.max_velocity
+                        assert ccw_target.position >= position_limit
+                    print(
+                        f"CCW target position={ccw_target.position}, velocity={ccw_target.velocity}"
+                    )
```
Comment (wvreeven) [inline tests/test_csc.py:456]: Is this print useful for the unit test or a remainder of debugging the code?

File: tests/test_mock_controller.py [hunk 6]
```diff
@@ -29,6 +29,7 @@
 
 from lsst.ts import salobj
 from lsst.ts import mtmount
+from lsst.ts import utils
```
Comment (wvreeven) [inline tests/test_mock_controller.py:32]: Same bit of nitpicking but wouldn't `from lsst.ts import salobj, mtmount, utils` make sense here?

File: tests/test_mock_controller.py [hunk 7]
```diff
@@ -513,12 +514,12 @@ async def test_tracking(self):
             self.assertTrue(device.enabled)
             self.assertTrue(device.tracking_enabled)
```
Comment (wvreeven) [inline tests/test_mock_controller.py:515]: Perhaps replace this with `assert`s as well?

File: tests/test_mock_controller.py [hunk 8]
```diff
@@ -584,7 +585,7 @@ async def test_move(self):
             self.assertTrue(device.enabled)
             self.assertFalse(device.tracking_enabled)
```
Comment (wvreeven) [inline tests/test_mock_controller.py:586]: Perhaps replace this with `assert`s as well?

File: tests/test_mock_devices.py [hunk 9]
```diff
@@ -384,7 +384,7 @@ async def check_axis_device(self, device):
         self.assertFalse(device.has_target)
```
Comment (wvreeven) [inline tests/test_mock_devices.py:384]: Perhaps replace this with an `assert` as well?

File: tests/test_mock_devices.py [hunk 10]
```diff
@@ -384,7 +384,7 @@ async def check_axis_device(self, device):
...
 
         # Do a point to point move
-        start_segment = device.actuator.path.at(salobj.current_tai())
+        start_segment = device.actuator.path.at(utils.current_tai())
         self.assertEqual(start_segment.velocity, 0)
```
Comment (wvreeven) [inline tests/test_mock_devices.py:388]: Perhaps replace this with an `assert` as well?

File: tests/test_mock_devices.py [hunk 11]
```diff
@@ -394,12 +394,12 @@ async def check_axis_device(self, device):
         await asyncio.sleep(0.1)  # give command time to start
         self.assertAlmostEqual(device.actuator.target.position, end_position)
         self.assertEqual(device.actuator.target.velocity, 0)
```
Comment (wvreeven) [inline tests/test_mock_devices.py:396]: Perhaps replace this with `assert`s as well? And please check the rest of the code in this file as well.

File: tests/test_csc.py (hunk 3 above)
Comment (r-owen) [reply tests/test_csc.py:413]: Good question. If something goes wrong it allows me to more quickly diagnose what the problem was. So I consider it part of the test.

File: tests/test_csc.py (hunk 4 above)
Comment (r-owen) [reply tests/test_csc.py:422]: Yes. The problem is that the system clock on Docker macOS has occasional small backward jumps. If a tracking command arrives with TAI <= previous commanded TAI the simulated tracking actuator rejects the command, which breaks the test. All that is required for these tests to pass reliably is that the commanded times be monotonically increasing.

Comment (wvreeven) [reply tests/test_csc.py:422]: Yes, I understood that. I was merely wondering if adding `0.001` always is enough and if that could possibly change in the future. I guess that if it does we will deal with it then.

File: tests/test_mock_controller.py [hunk 12]
```diff
@@ -620,7 +621,7 @@ async def test_move(self):
                     desired_value = end_position
                 else:
                     desired_value = 0
-                self.assertAlmostEqual(axis_telem[name], desired_value, msg=name)
+                assert axis_telem[name] == pytest.approx(desired_value)
 
 
 if __name__ == "__main__":
```
Comment (wvreeven) [inline tests/test_mock_controller.py:627]: I noticed you removed these lines from your other tests. Perhaps you could remove them from this one as well? I'll leave that for you to decide.

**Pull request merged**